neo4j/internal/bolt/hydrator.go

```diff
@@ -878,6 +885,20 @@ func parseNotifications(notificationsx []any) []db.Notification {
 	return notifications
 }
 
+func parseGqlStatusObjects(statuses []any) []db.GqlStatusObject {
+	var gqlStatusObjects []db.GqlStatusObject
+	if statuses != nil {
```

Comment:
It might work fine without the guard given how h.array() works. I implemented it like this to match the `parseNotifications` implementation: https://github.com/StephenCathcart/neo4j-go-driver/blob/016d5f1676960df0ced1b93defc328b6042e50e9/neo4j/internal/bolt/hydrator.go#L874-L876.

Reply:
TestKit tests still pass with the guard removed, happy to keep or remove the guard.

Reply:
leaving open for discussion